File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ret = [1] * len(nums)
        multi = 1

        for i in range(len(nums)):
            ret[i] = multi
            multi *= nums[i]

        multi = 1
        for i in range(len(nums) -1 , -1 ,-1):
            ret[i] *= multi
            multi *= nums[i]

        return ret

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] prefixSum = new int[nums.Length];
        int len = nums.Length;
        var multi = 1;


        for (int i = 0; i < len; i++)
        {
            prefixSum[i] = multi;
            multi *= nums[i];
        }
        multi =1;
        for (int i = len -1 ; i >= 0 ; i--)
        {
            prefixSum[i] *= multi;
            multi *= nums[i];
        }
    
        return prefixSum;
    }

    
}